src/frontend/core/executor/impl/TriangleCountExecutor.cpp

```diff
@@ -1727,17 +1725,12 @@ int TriangleCountExecutor::collectPerformaceData(PerformanceSQLiteDBInterface pe
         }
     }
 
-    start = time(0);
     performanceUtil.collectSLAResourceConsumption(placeList, graphId, command, category, masterIP, elapsedTime,
                                                   autoCalibrate);
-
     while (!workerResponded) {
-        if (time(0) - start == Conts::LOAD_AVG_COLLECTING_GAP) {
-            elapsedTime += Conts::LOAD_AVG_COLLECTING_GAP * 1000;
-            performanceUtil.collectSLAResourceConsumption(placeList, graphId, command, category, masterIP, elapsedTime,
-                                                          autoCalibrate);
-            start = start + Conts::LOAD_AVG_COLLECTING_GAP;
-        }
+        std::this_thread::sleep_for(std::chrono::milliseconds(Conts::LOAD_AVG_COLLECTING_GAP*1000));
+        elapsedTime += Conts::LOAD_AVG_COLLECTING_GAP * 1000;
```

Comment:
`std::this_thread::sleep_for(sleep_duration)` may sleep for more than the requested time*. But the `elapsedTime` is only incremented by the requested time. This time difference will keep accumulating in the loop. This can result in a significant deviation of `elapsedTime` value from the actual time elapsed, making the resource consumption metrics inaccurate as the time passes.

*reference: [en.cppreference.com/w/cpp/thread/sleep_for](https://en.cppreference.com/w/cpp/thread/sleep_for)
> This function may block for longer than `sleep_duration` due to scheduling or resource contention delays.

The reference also recommends,
> The standard recommends that a steady clock is used to measure the duration.